prometheus_hardware_exporter/collector.py

```diff
@@ -109,3 +112,261 @@ def fetch(self) -> List[Payload]:
+                )
+            elif current_item_unit == "%":
+                if sensor_data_item.get("Type") == "Fan":
+                    payloads.append(
+                        self._create_sensor_data_payload(sensor_data_item, "ipmi_fan_speed_ratio")
+                    )
+                else:
+                    payloads.append(
+                        self._create_sensor_data_payload(
+                            sensor_data_item, "ipmi_generic_sensor_value"
+                        )
+                    )
+            else:
+                payloads.append(
+                    self._create_sensor_data_payload(sensor_data_item, "ipmi_generic_sensor_value")
+                )
+
+        return payloads
+
+    def process(self, payloads: List[Payload], datastore: Dict[str, Payload]) -> List[Payload]:
+        """Process the payload if needed."""
+        return payloads
+
+    def _create_sensor_data_payload(
+        self, sensor_data_item: Dict[str, str], metric_name: str
+    ) -> Payload:
+        """Create a payload based on metric name and sensor data."""
+        if metric_name == "ipmi_generic_sensor_value":
+            return Payload(
+                name=metric_name,
+                labels=[
+                    sensor_data_item["ID"],
+                    sensor_data_item["Name"],
+                    sensor_data_item["State"],
+                    sensor_data_item["Units"],
+                    sensor_data_item["Event"],
+                    sensor_data_item["Type"],
+                ],
+                value=self._get_sensor_value_from_reading(sensor_data_item["Reading"]),
+            )
+
+        return Payload(
+            name=metric_name,
+            labels=[
+                sensor_data_item["ID"],
+                sensor_data_item["Name"],
+                sensor_data_item["State"],
+                sensor_data_item["Units"],
+                sensor_data_item["Event"],
+            ],
+            value=self._get_sensor_value_from_reading(sensor_data_item["Reading"]),
+        )
+
+    def _get_sensor_value_from_reading(self, reading: str) -> float:
+        """Get sensor value as a float from sensor reading.
+
+        Returns:
+            a floating point of the sensor reading, or 0.0
+        """
+        try:
+            return float(reading)
+        except ValueError:
+            logger.info("Changing sensor data value %s to 0.0", reading)
+            return 0.0
+
+
+class IpmiSelCollector(BlockingCollector):
+    """Collector for IPMI SEL data."""
+
+    ipmi_sel = IpmiSel()
+
+    @property
+    def specifications(self) -> List[Specification]:
+        """Define IPMI SEL metrics specs."""
+        return [
+            Specification(
+                name="ipmi_sel_state",
+                documentation="Event state from IPMI SEL entry.",
+                labels=["id", "date", "time", "name", "type", "event"],
```

Comment:
ID / date / time / event seems changes every time. ID will be a increment counter. I think we should collect the event in Loki instead of Prometheus.
Therefore, I also purpose to remove these labels to reduce the dimentions.